### DISCUSSION TITLE
Manage 20.0.0 Keycloak Operator from another Operator

### DISCUSSION THREAD

**Discussion by jaymberry** [keycloak/keycloak#13194] [Archive]
We are currently writing our own operator in Golang. When installing dependencies that are operators themselves, we need to add those operator CRDs to the schema through a golang lib. With keycloak 20.0.0 we cannot do this because it is written in Java. 

How can we install Keycloak 20.0.0 operator written in Java with its CRDs with our Operator written Golang?

**Comment by jaymberry** [comment]
Hi, I am just wondering if there is anyone to help with this. Or is there an estimate to when 20.0.0 will be able to provide this functionality to the larger operator community?

Thanks

**Comment by vmuzikar** [comment]
Hello, thank you for your interested in the new Operator!

As you correctly mentioned, the Operator is written in Java. That decision was made to provide closer integration with Keycloak itself and better alignment with our devs and community skill set. The obvious trade-off is that we don't the CRDs representations written in Go and we don't plan to add such support at the moment.

However, a possible workaround could be [dynamic resources](https://itnext.io/generically-working-with-kubernetes-resources-in-go-53bce678f887) that the Go K8s Client supports for cases like this.

Let me know if that works for your use-case.

**Reply by jaymberry** [reply]
Hello, thank you. This works for our use case.
